### ISSUE TITLE
[REQUEST] Continuation after end of MS support (Mar 2025)

### ISSUE THREAD

**Issue by robbely** [MustardChef/WSABuilds#330]
### Describe your request

Is this project 100% supplied by the MS files, or is there a chance of continued development after official support ends next year?

**Comment by NebularNerd** [comment]
It does use files from MS which contain the main Android system, I guess it would break down into:

- Will the files be left sitting on MS Servers or will they instantly delete them?
- Alternatively, could they be hosted here on the repo, or archive.org? (doubtful as they are MS property)
- Could @MustardChef / we the community make an Android build that could take the place of the MS build?

Obviously there big thing is future proofing. At the moment the current WSA Android is running a version that Google supports, but with time it will become outdated.

**Comment by html5syt** [comment]
e...
I have a immature idea: if we can change the system img when the new Android release, and maybe we can decompilation the app bundle.
Is it can? I don't want to see this so useful program be archived:(

**Comment by cloewen8** [comment]
I'm onboard to contribute where possible to get a community-run Android build. This is a valuable tool to me and based on what I've seen, it is for a lot of users.

**Comment by NebularNerd** [comment]
I've not done any modern Android builds, but in the distant past, I used to make a custom rom for a Chinese tablet so I have a little (but outdated) experience in this area. The main issues are compiling from source, drivers, how to package it up, and obviously trust (We've likely all seen Linus's video about the cheap TV Box riddled with malware). But it would be cool to see if we could keep it alive.

**Comment by Snowlights2022** [comment]
> e... I have a immature idea: if we can change the system img when the new Android release, and maybe we can decompilation the app bundle. Is it can? I don't want to see this so useful program be archived:(

 I am concerned that these bundles will lose their source of download (or rather redistribution) after MS end of support by some reason :(

**Comment by html5syt** [comment]
> > e... I have a immature idea: if we can change the system img when the new Android release, and maybe we can decompilation the app bundle. Is it can? I don't want to see this so useful program be archived:(
> 
> I am concerned that these bundles will lose their source of download (or rather redistribution) after MS end of support by some reason :(

if we can redirect them to Github or other stable places?
And the WSA have good communications to Windows.Maybe we can extract the apps that communicate between Android and Windows? And maybe we can use some images like Genymobile or simulator inside Android Studio?

**Comment by html5syt** [comment]
I think Microsoft not encrypted the MSIX and images.Because it's a not very popular software before.

**Comment by miles5600** [comment]
upvote this in the feedback hub to get Microsoft's attention

https://aka.ms/AAqk69c

**Comment by NebularNerd** [comment]
> upvote this in the feedback hub to get Microsoft's attention
> 
> https://aka.ms/AAqk69c

No luck for me, is this for Insiders only?
![image](https://github.com/MustardChef/WSABuilds/assets/8470449/b8c98b50-428d-4956-b0be-7ef91222b4bd)

**Comment by cloewen8** [comment]
Still invested, even in this outside of Feedback (I don't like that Feedback requires another install).

**Comment by StefanG3D** [comment]
WSA+Tencent App Store rumor
https://winfuture.de/news,143107.html

**Comment by realgooseman** [comment]
Probably will be targeted to chinese market only. We'll be stuck with those crappy proprietary emulators on Windows (Bluestacks, Nox, LD, MeMu, ...)

**Comment by NebularNerd** [comment]
> WSA+Tencent App Store rumor https://winfuture.de/news,143107.html

Interesting. Assuming it's WSA then it should allow for the MustardChef builds to continue in some form. It should be possible to remove Chinese-centric apps, much like de-Amazoning the current WSA system.

**Comment by html5syt** [comment]
To tell the truth, I'm a Chinese user of WSA, Tencent maybe is a good company, but if you search online, you will found that Tencent made many terrible things like bad support, various membership systems, Inhumane and intrusive advertising and so on.
For another reason, china is setting off a wave of self-developed systems and want to completely deprecated Android. But it will make many difficulties to install and use apps from abroad to Chinese users.

**So maybe it's a bad news not only to Chinese users, but also to all users.**

**Comment by angaratosurion** [comment]
> WSA+Tencent App Store rumor https://winfuture.de/news,143107.html

if its true will jsut need to downlaod them from servers they will be hsoted .
Get rid of possible country limitations , get rid of the crap tencent adds and also translate everything too english in case it doesnt have english language

**Comment by angaratosurion** [comment]
is it allowed by law to reverse engineer the wsa and re write it  in a way itdoesnt infringes MS  copyright ?

**Comment by NebularNerd** [comment]
> is it allowed by law to reverse engineer the wsa and re write it in a way itdoesnt infringes MS copyright ?

The main issue of copyright would revolve around the WSA part itself. In theory any Android x86/x64 could run under WSA (think Lineage and similar), much like almost any *nix can run under WSL. 

The WSA parts are the bits that keep both OS's talking to each other and pass hardware I/O for graphics, networking and such (much like VMWare Tools does) and that's the sticking point.

**Comment by angaratosurion** [comment]
> > is it allowed by law to reverse engineer the wsa and re write it in a way itdoesnt infringes MS copyright ?
> 
> The main issue of copyright would revolve around the WSA part itself. In theory any Android x86/x64 could run under WSA (think Lineage and similar), much like almost any *nix can run under WSL.
> 
> The WSA parts are the bits that keep both OS's talking to each other and pass hardware I/O for graphics, networking and such (much like VMWare Tools does) and that's the sticking point.

Yeah the wsa wouldn't need to be reverse enginered in order to be maintained?

**Comment by AlexDev404** [comment]
> > upvote this in the feedback hub to get Microsoft's attention
> > https://aka.ms/AAqk69c
> 
> No luck for me, is this for Insiders only? ![image](https://private-user-images.githubusercontent.com/8470449/334010853-b8c98b50-428d-4956-b0be-7ef91222b4bd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQzNzcxMDgsIm5iZiI6MTcyNDM3NjgwOCwicGF0aCI6Ii84NDcwNDQ5LzMzNDAxMDg1My1iOGM5OGI1MC00MjhkLTQ5NTYtYjBiZS03ZWY5MTIyMmI0YmQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDgyMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA4MjNUMDEzMzI4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NDlhYjUwYThhODFjYzk1MjY2ZmFmOGYzZjkwN2JjNGNlNjg5OWZkZTkwOTZlNDQ1ZTg2YWVkNjFmZDdhNzhkZSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.MbwPMrn2uHaFp7vHSknmbSRixrsr-YH4MBGKCoMc8j0)

No, haha you just need to sign in with a Microsoft account first.

**Comment by NebularNerd** [comment]
> > > upvote this in the feedback hub to get Microsoft's attention
> > > https://aka.ms/AAqk69c
> > 
> > 
> > No luck for me, is this for Insiders only? ![image](https://private-user-images.githubusercontent.com/8470449/334010853-b8c98b50-428d-4956-b0be-7ef91222b4bd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQzNzcxMDgsIm5iZiI6MTcyNDM3NjgwOCwicGF0aCI6Ii84NDcwNDQ5LzMzNDAxMDg1My1iOGM5OGI1MC00MjhkLTQ5NTYtYjBiZS03ZWY5MTIyMmI0YmQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDgyMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA4MjNUMDEzMzI4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NDlhYjUwYThhODFjYzk1MjY2ZmFmOGYzZjkwN2JjNGNlNjg5OWZkZTkwOTZlNDQ1ZTg2YWVkNjFmZDdhNzhkZSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.MbwPMrn2uHaFp7vHSknmbSRixrsr-YH4MBGKCoMc8j0)
> 
> No, haha you just need to sign in with a Microsoft account first.

Typically poor programmed MS app 😂 after signing in it does not show me post I should be upvoting. I assume we are all going with this one:
![image](https://github.com/user-attachments/assets/b851dc0f-f1a1-4eeb-8e29-a7783d831390)

**Comment by AlexDev404** [comment]
> > > > upvote this in the feedback hub to get Microsoft's attention
> > > > https://aka.ms/AAqk69c
> > > 
> > > 
> > > No luck for me, is this for Insiders only? ![image](https://private-user-images.githubusercontent.com/8470449/334010853-b8c98b50-428d-4956-b0be-7ef91222b4bd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQzNzcxMDgsIm5iZiI6MTcyNDM3NjgwOCwicGF0aCI6Ii84NDcwNDQ5LzMzNDAxMDg1My1iOGM5OGI1MC00MjhkLTQ5NTYtYjBiZS03ZWY5MTIyMmI0YmQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDgyMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA4MjNUMDEzMzI4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NDlhYjUwYThhODFjYzk1MjY2ZmFmOGYzZjkwN2JjNGNlNjg5OWZkZTkwOTZlNDQ1ZTg2YWVkNjFmZDdhNzhkZSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.MbwPMrn2uHaFp7vHSknmbSRixrsr-YH4MBGKCoMc8j0)
> > 
> > 
> > No, haha you just need to sign in with a Microsoft account first.
> 
> Typically poor programmed MS app 😂 after signing in it does not show me post I should be upvoting. I assume we are all going with this one: ![image](https://private-user-images.githubusercontent.com/8470449/360840218-b851dc0f-f1a1-4eeb-8e29-a7783d831390.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQ3MDU2NzgsIm5iZiI6MTcyNDcwNTM3OCwicGF0aCI6Ii84NDcwNDQ5LzM2MDg0MDIxOC1iODUxZGMwZi1mMWExLTRlZWItOGUyOS1hNzc4M2Q4MzEzOTAucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDgyNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA4MjZUMjA0OTM4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YzlhMmJiYTA3OWU4YjM5YjkxZTdmNzdkNTI4MjY0MjdiNzgwNDFmODY5YmU1NjRkY2YzMDI4NmMyM2NiOWQ4MCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.fBKxo8dEbFnmOo_W1cahNUomz6cww3F7uHxMFZVhJyM)

They be stressing their engineers out too much, lol

**Comment by KatxCryn** [comment]
https://github.com/MustardChef/WSABuilds/issues/330#issuecomment-2132402105

**Comment by Andrew-J-Larson** [comment]
So, now that WSA is officially no longer being supported (as of yesterday), what is the verdict on supporting the unofficial WSA builds?

**Comment by HamzaMalik3** [comment]
I want to download for Windows 10 but it will not downloaded stop at 40 %  no response comes from server. just WSAMagiskDelta-Windows_10_2307.40000.5.0 is ready to download not others but this error is severe. Please fix it or tell me how to download it. why it is not available on any other website other then GitHub

**Comment by Andrew-J-Larson** [comment]
> I want to download for Windows 10 but it will not downloaded stop at 40 % no response comes from server. just WSAMagiskDelta-Windows_10_2307.40000.5.0 is ready to download not others but this error is severe. Please fix it or tell me how to download it. why it is not available on any other website other then GitHub

Wrong place for this, this should have been a new issue.
